project-minimization-plugin/src/main/kotlin/org/plan/research/minimization/plugin/settings/AppSettingsComponent.kt

```diff
@@ -207,12 +236,59 @@ class AppSettingsComponent {
             .addSeparator()
             .addLabeledComponent(JBLabel("Transformations:"), createTransformationPanel(), 1, false)
             .addSeparator()
+            .addComponent(JBLabel("Add here directories/files of your project if you are sure that they are important or can not be minimized").apply {
+                border = BorderFactory.createEmptyBorder(10, 0, 10, 0)
+            })
+            .addComponent(pathsPanel)
+            .addSeparator()
             .addComponentFillVertically(JPanel(), 0)
             .panel
 
         updateUIState()
     }
 
+    private fun createPathPanel(): JPanel {
+        pathTable.columnModel.getColumn(0).apply {
+            preferredWidth = 700  // Width of first column
+        }
+
+        val toolbarDecoratorPanel = ToolbarDecorator.createDecorator(pathTable)
+            .disableUpAction()
+            .disableDownAction()
+            .setAddAction {
+                val chosenFiles = FileChooser.chooseFiles(fileChooserDescriptor, null, null)
+                for (file in chosenFiles) {
+                    val path = file.path
```

Comment:
Also consider to use relative paths here

Reply:
You can use something like `file.toNioPath().relativeTo(project.guessProjectDir().toNioPath())`